Classes/RequestMethod/CurlPostWithProxy.php

```diff
@@ -63,14 +63,17 @@ public function injectSettings(array $settings)
             throw new \Exception("Invalid configuration, the Wegmeister.Recaptcha.httpProxy option has to be a string.");
         }
 
-        $httpProxyParts = explode(':', $settings['httpProxy']);
-        if(count($httpProxyParts) !== 2 || (int)$httpProxyParts[1] === 0){
+        $httpProxy = $settings['httpProxy'];
+        $this->emitHttpProxyRetrieved($httpProxy);
+
+        $formatIsValid = preg_match('@(.*):(\d{1,5})@', $httpProxy, $httpProxyParts);
```

Comment:
I think, we need to make sure no other values are before or after the string. So IMO `'@^(.*):(\d{1,5})$@'` should be used.